contrib/database_statistics.lua

```diff
@@ -0,0 +1,441 @@
+gettext.bindtextdomain("database_statistics",dt.configuration.config_dir.."/lua/")
+
+local function _(msgid)
+    return gettext.dgettext("database_statistics", msgid)
+end
+
+local labeldiscusage = dt.new_widget("label")
+{
+     label = _('Disc usage:\t\t\t\t') .. count_discusage,
+     ellipsize = "start",
+     halign = "start"
+}
+
+local labelfilmrolls = dt.new_widget("label")
+{
+     label = _('Filmrolls:\t\t\t\t\t') .. count_filmrolls,
+     ellipsize = "start",
+     halign = "start"
+}
+
+
+local labelimages = dt.new_widget("label")
+{
+      label = _('Images:\t\t\t\t\t') .. count_images,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local separator_1 = dt.new_widget("separator")
+{}
+
+
+local rating_1_label = dt.new_widget("label")
+{ 
+      label = _('Images with 1 star:\t\t\t') .. count_rating_1,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local rating_2_label = dt.new_widget("label")
+{ 
+      label = _('Images with 2 stars:\t\t\t') .. count_rating_2,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local rating_3_label = dt.new_widget("label")
+{ 
+      label = _('Images with 3 stars:\t\t\t') .. count_rating_3,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local rating_4_label = dt.new_widget("label")
+{ 
+      label = _('Images with 4 stars:\t\t\t') .. count_rating_4,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local rating_5_label = dt.new_widget("label")
+{ 
+      label = _('Images with 5 stars:\t\t\t') .. count_rating_5,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local rating_0_label = dt.new_widget("label")
+{ 
+      label = _('Images without stars:\t\t') .. count_rating_0,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local separator_2 = dt.new_widget("separator")
+{}
+
+
+local colorlabelred = dt.new_widget("label")
+{ 
+      label = _('Images with red labels:\t\t') .. count_red,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local colorlabelyellow = dt.new_widget("label")
+{
+      label = _('Images with yellow labels:\t\t') .. count_yellow,
+      ellipsize = "start",
+      halign = "start",
+}
+
+
+local colorlabelgreen = dt.new_widget("label")
+{
+      label = _('Images with green labels:\t\t')  .. count_green,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local colorlabelblue = dt.new_widget("label")
+{
+      label = _('Images with blue labels:\t\t') .. count_blue,
+      ellipsize = "start",
+      halign = "start",
+}
+
+
+local colorlabelmag = dt.new_widget("label")
+{
+      label = _('Images with magenta labels:\t') .. count_magenta,
+      ellipsize = "start",
+      halign = "start",
+}
+
+local nocolorlabels = dt.new_widget("label")
+{
+      label = _('Images without labels:\t\t') .. count_without,
+      ellipsize = "start",
+      halign = "start",
+}
+
+
+local function checkIfBinExists(bin)
```

Comment:
We have a new utility class for this function, that you could use. Have a look into the geoToolbox.lua script:

local df = require "lib/dtutils.file"
df.check_if_bin_exists("PROGRAMM")